client/utils/utils.go

```diff
@@ -0,0 +1,74 @@
+package utils
+
+import (
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/context"
+	"github.com/cosmos/cosmos-sdk/client/keys"
+	cryptokeys "github.com/cosmos/cosmos-sdk/crypto/keys"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authctx "github.com/cosmos/cosmos-sdk/x/auth/client/context"
+)
+
+// GetPassphraseFromStdin gets the passphrase from STDIN. An error is returned
+// if getting input fails.
+func GetPassphraseFromStdin(name string) (string, error) {
+	buf := client.BufferStdin()
+	prompt := fmt.Sprintf("Passphrase to sign with '%s':", name)
+
+	return client.GetPassword(prompt, buf)
+}
+
+// SendTx implements a handler that facilitates sending a series of messages in
+// a signed transaction given a TxContext and a QueryContext. It ensures that
+// the account has a proper number and sequence set. In addition, it builds and
+// signs a transaction with the supplied messages. Finally, it broadcasts the
+// signed transaction to a node.
+func SendTx(txCtx authctx.TxContext, queryCtx context.QueryContext, from []byte, name string, msgs []sdk.Msg) error {
+	if txCtx.AccountNumber == 0 {
+		accNum, err := queryCtx.GetAccountNumber(from)
```

Comment:
When rebasing, can you be sure to add the "EnsureAccountExists" function?

Reply:
@ValarDragon why is that? `SendTx` does what `EnsureAccountExists` virtually did. Do you want me to rip that out and call it `EnsureAccountExists` which `SendTx` would call?

Reply:
I see what you mean. Yeah I think I can leverage something similar 👍

Reply:
I'll add a method to `QueryContext#EnsureAccountExists`. This will get called before the any signing even happens.